### ISSUE TITLE
[Feature Request] Enable parameter builders and executers to determine if application is public client or confidential client

### ISSUE THREAD

**Issue by trwalke** [AzureAD/microsoft-authentication-library-for-dotnet#3334]
**Is your feature request related to a problem? Please describe.**
Enable parameter builders and and executers to determine if application is cublic client or confidential client
See https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/3316/files#r876471513

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by jmprieur** [comment]
I don't understand the why, @trwalke ?

**Comment by bgavrilMS** [comment]
Agreed with @jmprieur, we will need to separate AcquireTokenSilent instead into PCA and CCA versions (possibly in MSAL5).
